src/com/owncloud/android/files/services/FileUploadService.java

```diff
@@ -0,0 +1,1359 @@
+ * 
+ * @author LukeOwncloud
+ * 
+ */
+@SuppressWarnings("unused")
+public class FileUploadService extends Service implements OnDatatransferProgressListener {
+
+    private volatile Looper mServiceLooper;
+    private volatile ServiceHandler mServiceHandler;
+    private ExecutorService uploadExecutor;
+
+    public FileUploadService() {
+        super();
+    }
+
+    private static final String UPLOAD_FINISH_MESSAGE = "UPLOAD_FINISH";
+    public static final String EXTRA_UPLOAD_RESULT = "RESULT";
+    public static final String EXTRA_REMOTE_PATH = "REMOTE_PATH";
+    public static final String EXTRA_OLD_REMOTE_PATH = "OLD_REMOTE_PATH";
+    public static final String EXTRA_OLD_FILE_PATH = "OLD_FILE_PATH";
+    public static final String ACCOUNT_NAME = "ACCOUNT_NAME";
+
+    public static final String KEY_FILE = "FILE";
+    public static final String KEY_LOCAL_FILE = "LOCAL_FILE";
+    public static final String KEY_REMOTE_FILE = "REMOTE_FILE";
+    public static final String KEY_MIME_TYPE = "MIME_TYPE";
+    
+    /**
+     * Call this Service with only this Intent key if all pending uploads are to be retried.
+     */
+    private static final String KEY_RETRY = "KEY_RETRY";
+    /**
+     * Call this Service with KEY_RETRY and KEY_RETRY_REMOTE_PATH to retry
+     * download of file identified by KEY_RETRY_REMOTE_PATH.
+     */
+    private static final String KEY_RETRY_REMOTE_PATH = "KEY_RETRY_REMOTE_PATH";    
+    /**
+     * {@link Account} to which file is to be uploaded.
+     */
+    public static final String KEY_ACCOUNT = "ACCOUNT";
+    /**
+     * Set whether single file or multiple files are to be uploaded. Value must be of type {@link UploadSingleMulti}.
+     */
+    public static final String KEY_UPLOAD_TYPE = "UPLOAD_TYPE";
+    /**
+     * Set to true if remote file is to be overwritten. Default action is to upload with different name. 
+     */
+    public static final String KEY_FORCE_OVERWRITE = "KEY_FORCE_OVERWRITE";
+    /**
+     * Set to true if remote folder is to be created if it does not exist.
+     */
+    public static final String KEY_CREATE_REMOTE_FOLDER = "CREATE_REMOTE_FOLDER";
+    /**
+     * Set to true if upload is to performed only when connected via wifi.
+     */
+    public static final String KEY_WIFI_ONLY = "WIFI_ONLY";
+    /**
+     * Set to true if upload is to performed only when phone is being charged.
+     */
+    public static final String KEY_WHILE_CHARGING_ONLY = "KEY_WHILE_CHARGING_ONLY";
+    /**
+     * Set to future UNIX timestamp. Upload will not be performed before this timestamp. 
+     */
+    public static final String KEY_UPLOAD_TIMESTAMP= "KEY_UPLOAD_TIMESTAMP";
+    
+    public static final String KEY_LOCAL_BEHAVIOUR = "BEHAVIOUR";
+
+    /**
+     * Describes local behavior for upload.
+     */
+    public enum LocalBehaviour {
+        /**
+         * Creates a copy of file and stores it in tmp folder inside owncloud
```

Comment:
You are welcome! I thought if I dig into the code to find out what it does, I should share my "findings" with everybody.